File: frontend/types/subscription.ts
import { Plan } from './plan';
import { Organization } from './organization';

export type SubscriptionStatus = 'ACTIVE' | 'CANCELLED' | 'EXPIRED' | 'PENDING' | 'SUSPENDED';

export interface Subscription {
  id: string;
  organization: Organization;
  plan: Plan;
  status: SubscriptionStatus;
  startDate: string;
  endDate: string;
  autoRenew: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface SubscriptionsResponse {
  subscriptions: {
    success: boolean;
    message: string;
    subscriptions: Subscription[];
  };
}

export interface SubscriptionResponse {
  subscription: {
    success: boolean;
    message: string;
    subscription: Subscription;
  };
}

File: backend/app/models/subscription.py
from datetime import datetime
from typing import Optional
from app.schemas.subscription import SubscriptionStatus
import strawberry

@strawberry.type
class DBSubscription:
    _id: str
    organizationId: str
    planId: str
    status: SubscriptionStatus
    startDate: datetime
    endDate: datetime
    autoRenew: bool
    createdAt: datetime
    updatedAt: datetime